Update route_helpers

s-espinosa opened the pull request:
@Carmer @case-eee 

Here's my draft of the route_helper lesson. The link for BookShelf hasn't yet been updated, but otherwise I think it's in decent shape. 

I know that we had discussed not overwriting the existing file unless it was to make additions. I think this falls into this case, so this version is overwriting the previous version of the lesson. Let me know if you think it would be better to create a new/separate file and I can update my PR.

Let me know what you think!

Carmer commented on ruby_02-web_applications_with_ruby/outlines/route_helpers.md:
````diff
@@ -9,66 +9,87 @@ tags: routes, helpers
-  ```html
-    <a href="/artists/<%= artist.id %>"><%= artist.name %></a>
-  ```
+```erb
````
I think the intent before was to get them to generate the idea of using link_to on their own, instead of being given the solution. Let's see what @case-eee has to say as I think she made that change.

case-eee replied:
I like the idea of giving them a few minutes to do some research and see if they can implement it on their own 👍

Carmer commented:
Looks good @s-espinosa .. I think this falls in the category of modification of the lesson plan and it isn't destructive in a way that you loose a portion of the original content, so I think using the same lesson plan is fine. Thanks for considering that. 

I made only one comment on the code. It's a minor thing and I could go either way. I tagged @case-eee to see if she has more of an opinion on the matter.

case-eee commented on ruby_02-web_applications_with_ruby/outlines/route_helpers.md:
```diff
@@ -9,66 +9,87 @@ tags: routes, helpers
+In pairs, one of you write the CRUD routes for `song`, one for `phone`. If you're in a group of three, have the third person write the routes for `sticker`.
 
-##### Do you notice any repetition or patterns here?
+With your partner, do you notice any repetition or patterns in the routes that you wrote?
```
This is something they should absolutely be able to answer given that they have already seen the http verb + path name + action table and we've been reiterating it. 

Is there an additional question (or can we re-phrase this one) that's both review but might stretch them a little?

case-eee commented on ruby_02-web_applications_with_ruby/outlines/route_helpers.md:
```diff
@@ -9,66 +9,87 @@ tags: routes, helpers
+
+  * Song
+  * Phone
+  * Sticker
```
Is three too many here?

case-eee commented on ruby_02-web_applications_with_ruby/outlines/route_helpers.md:
```diff
@@ -9,66 +9,87 @@ tags: routes, helpers
+* What are the headers of the table that it gives you?
+* With your partner, take a look at the entries in the table that `rake routes` gives you and fill out the table below in your notebook or on your computer.
+
+|`rake routes` Table Heading|Exampe Entry|Potential Definition|
```
This might be a tad confusing since you've transposed the columns a bit (I could be completely wrong though), but I really like this activity of having them diagram this info

case-eee commented:
@s-espinosa I think this looks good 👍  Like we discussed, these are a lot of things to think about - not necessarily implement. Also - is there some independent practice time built in here?

s-espinosa commented:
@Carmer @case-eee Thanks for the feedback! I updated to include the following:

* New warmup - didn't realize they had already seen `rake routes` at this point. Updated with an example that will demonstrate rails ability to handle odd plural forms, and asked the students to attempt to create the route on their own.
* Reduced the number of resources for the students to create. Agree we can probably get just as much with a single example.
* Swapped the table orientation to match what `rake routes` gives you.
* Gave students the opportunity to figure out link helpers on their own.
* Added some independent work: creating `show`/`index` views for `song. Even if they can't finish in the time they have left, should give them an opportunity to practice the whole process.

Let me know what you think!

case-eee commented:
I think this is great. I know we just spoke about a few more ideas so let's touch base later this afternoon about it!

s-espinosa commented:
@Carmer @case-eee Updated based on where I think the students are currently. I also checked the link for BookShelf and it still works. Let's discuss briefly tomorrow morning and then I can merge.

Carmer commented:
Sounds good @s-espinosa . With this lesson I tend to dive into the output of rake routes and drill/question them about each piece of the output - particularly the prefix that gives them `_path` and `_url` helpers.

case-eee commented:
Sounds good!

On Tuesday, August 30, 2016, Andrew Carmer <notifications@github.com> wrote:

> Sounds good @s-espinosa <https://github.com/s-espinosa> . With this
> lesson I tend to dive into the output of rake routes and drill/question
> them about each piece of the output - particularly the prefix that gives
> them _path and _url helpers.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/turingschool/lesson_plans/pull/242#issuecomment-243661361>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGM9JQQXIaGwHs1V4jK3NMhDWLllaFCrks5qlQ96gaJpZM4JuNr8>
> .
>
